### PR TITLE
Add swipe to hide API to bottom sheet

### PR DESCRIPTION
**Pull request by lcapkovic**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

This PR adds a new `allowsSwipeToHide` bool prop to `BottomSheetController` which allows the user to drag the sheet below its collapsed state and fully hide it. The default value is `false`, which preserves the old behavior.

I also added an example in the demo controller, where every transient sheet now has this enabled.

### Verification

https://user-images.githubusercontent.com/3610850/190024741-bde61a5c-1cd0-4f02-b7c1-850302136737.mov

https://user-images.githubusercontent.com/3610850/190027691-b1c8b7b4-1e06-443b-b006-1cb3211eca1b.mov

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [x] Different resolutions (1x, 2x, 3x)
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [x] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1242)

### CODE REVIEW

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/BottomSheetDemoController.swift [hunk 1]
```diff
@@ -85,11 +85,13 @@ class BottomSheetDemoController: UIViewController {
         let sheetContentView = UIView()
 
         let secondarySheetController = BottomSheetController(expandedContentView: sheetContentView)
+        secondarySheetController.delegate = self
         secondarySheetController.collapsedContentHeight = 250
         secondarySheetController.isHidden = true
         secondarySheetController.shouldAlwaysFillWidth = false
         secondarySheetController.shouldHideCollapsedContent = false
         secondarySheetController.isFlexibleHeight = true
+        secondarySheetController.allowsSwipeToHide = true
```
Comment (harrieshin) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/BottomSheetDemoController.swift:94]: we should either have a toggle or clear description what is the behavior of this seconarydsheetcontroller in the demo app itself for future readers.

File: ios/FluentUI/Bottom Sheet/BottomSheetController.swift [hunk 2]
```diff
@@ -198,6 +198,9 @@ public class BottomSheetController: UIViewController {
         }
     }
 
+    /// When enabled, users will be able to move the sheet to the hidden state by swiping down.
+    @objc open var allowsSwipeToHide: Bool = false
```
Comment (harrieshin) [inline ios/FluentUI/Bottom Sheet/BottomSheetController.swift:202]: should we be explicit this isn't a thing in iPad?

Comment (lcapkovic) [reply ios/FluentUI/Bottom Sheet/BottomSheetController.swift:202]: It works the same way on iPad when it's enabled.

Comment (harrieshin) [reply ios/FluentUI/Bottom Sheet/BottomSheetController.swift:202]: i thought bottomsheet was a floating bar in ipad?

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/BottomSheetDemoController.swift (hunk 1 above)
Comment (lcapkovic) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/BottomSheetDemoController.swift:97]: These are the sheets that show when you tap the "show transient sheet" button in the demo app. I added an explanation in a comment.

File: ios/FluentUI/Bottom Sheet/BottomSheetController.swift (hunk 2 above)
Comment (lcapkovic) [reply ios/FluentUI/Bottom Sheet/BottomSheetController.swift:202]: No only bottom commanding behaves that way. Bottom sheets work on all devices.

**Review by harrieshin — APPROVED** [review]

**Pull request merged**